### ISSUE TITLE
Retribution opener and general tweaks...

### ISSUE THREAD

**Issue by Retridin** [Hekili/hekili#320]
I quite am new with Hekili and it seems nice, but there are few things that bugs me and I would like to know how if in anyway I can change/alter somethings...

1. The opener, when Execution sentence, Hammer of Wrath, Wake of Ashes and Crusade is chosen
- At the moment Hekili is suggesting sequence like:
Blade of Justice
Judgment
Crusade
Execution Sentence
Wake of Ashes
Templar's Verdict
Blade of Justice
Judgement
Templar's Verdict
...

- When it should be more like:
Blade of Justice
Judgment
Crusade
Templar's Verdict
Wake of Ashes
Templar's Verdict
Hammer of Wrath
Execution Sentence
...

That way you get the full benefit from ES, when you have closer to 10 or so stacks of Crusade.

Also really I would like to change is the opener with Righteous Verdict -talent.

- Now the opener is somewhere around:
Blade of Justice
Judgment
Crusade
Hammer of Wrath
Wake of Ashes
Templar's Verdict
Blade of Justice
Judgement
Templar's Verdict

- When it should be more like:
Blade of Justice
Judgment
Crusade
Templar's Verdict
Hammer of Wrath
Blade of Justice (Crusader strike)
Templar's Verdict
Wake of Ashes
Templar's Verdict
...

Basically ALWAYS first cast after Crusade (Avengin Wrath) should be Templar's Verdict, and after that HoW... So TW > HoW

**Comment by Hekili** [comment]
Sure, the way to work on that is to take the existing sim profile and modify it until it does the thing you want, then link the result to me.

**Comment by Hekili** [comment]
(You can also edit the priority directly, in the /hekili > Priorities section.)

**Comment by Retridin** [comment]
Oh, thanks for quick reply! :)

Hmm, edit priority directly.. Is it this place? ..attached screenshot.
![Hekili_priorities](https://user-images.githubusercontent.com/63414414/78917432-851ad500-7a97-11ea-8b47-62fe792062f2.JPG)

**Comment by Hekili** [comment]
If you're wanting to edit it in SimulationCraft style, sure.  Or you can just edit the action lists that were imported from that profile in the next tab.  But...  You're really mostly on your own there.

**Comment by Retridin** [comment]
Not sure what you mean, but here is Example from SimC..

From SimC: normal sequence:

Pre	precombat	0	flask	Galahads	0.0/5: 0% holy_power	
Pre	precombat	1	food	Galahads	0.0/5: 0% holy_power	
Pre	precombat	2	augmentation	Galahads	0.0/5: 0% holy_power	
Pre	precombat	4	potion	Fluffy_Pillow	0.0/5: 0% holy_power	
0:00.000	default	7	auto_attack	Fluffy_Pillow	0.0/5: 0% holy_power	potion_of_focused_resolve
0:00.000	cooldowns	B	shield_of_vengeance	Galahads	0.0/5: 0% holy_power	potion_of_focused_resolve
0:01.197	generators	H	blade_of_justice	Fluffy_Pillow	0.0/5: 0% holy_power	bloodlust, shield_of_vengeance, potion_of_focused_resolve
0:02.118	generators	I	judgment	Fluffy_Pillow	2.0/5: 40% holy_power	bloodlust, shield_of_vengeance, strife, potion_of_focused_resolve
0:03.040	cooldowns	C	crusade	Fluffy_Pillow	3.0/5: 60% holy_power	bloodlust, shield_of_vengeance, deadly_navigation, strife, potion_of_focused_resolve
0:03.962	generators	J	hammer_of_wrath	Fluffy_Pillow	3.0/5: 60% holy_power	bloodlust, crusade, shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife, potion_of_focused_resolve
0:04.857	finishers	E	templars_verdict	Fluffy_Pillow	4.0/5: 80% holy_power	bloodlust, crusade, shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife(2), potion_of_focused_resolve
0:05.751	generators	G	wake_of_ashes	Fluffy_Pillow	1.0/5: 20% holy_power	bloodlust, crusade(4), shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife(2), potion_of_focused_resolve
0:06.575	finishers	E	templars_verdict	Fluffy_Pillow	5.0/5: 100% holy_power	bloodlust, crusade(4), shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife(2), potion_of_focused_resolve
0:07.398	generators	H	blade_of_justice	Fluffy_Pillow	2.0/5: 40% holy_power	bloodlust, crusade(7), shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife(2), potion_of_focused_resolve
0:08.160	generators	J	hammer_of_wrath	Fluffy_Pillow	4.0/5: 80% holy_power	bloodlust, crusade(7), shield_of_vengeance, avengers_might, deadly_navigation, strife(2), potion_of_focused_resolve
...

In that one you are already wasting one HP using Wake of Ashes at 1HP which is generated by Hammer of Wrath.


When it absolutely should be reversed those Templar´s Verdict and Hammer of Wrath, so you dont waste ANY HP.

Pre	precombat	0	flask	Galahads	0.0/5: 0% holy_power	
Pre	precombat	1	food	Galahads	0.0/5: 0% holy_power	
Pre	precombat	2	augmentation	Galahads	0.0/5: 0% holy_power	
Pre	precombat	4	potion	Fluffy_Pillow	0.0/5: 0% holy_power	
0:00.000	default	7	auto_attack	Fluffy_Pillow	0.0/5: 0% holy_power	potion_of_focused_resolve
0:00.000	cooldowns	B	shield_of_vengeance	Galahads	0.0/5: 0% holy_power	potion_of_focused_resolve
0:01.197	generators	H	blade_of_justice	Fluffy_Pillow	0.0/5: 0% holy_power	bloodlust, shield_of_vengeance, potion_of_focused_resolve
0:02.118	generators	I	judgment	Fluffy_Pillow	2.0/5: 40% holy_power	bloodlust, shield_of_vengeance, strife, potion_of_focused_resolve
0:03.040	cooldowns	C	crusade	Fluffy_Pillow	3.0/5: 60% holy_power	bloodlust, shield_of_vengeance, deadly_navigation, strife, potion_of_focused_resolve
0:03.962	finishers	E	templars_verdict	Fluffy_Pillow	4.0/5: 80% holy_power	bloodlust, crusade, shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife(2), potion_of_focused_resolve
0:04.857	generators	G	wake_of_ashes	Fluffy_Pillow	1.0/5: 20% holy_power	bloodlust, crusade(4), shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife(2), potion_of_focused_resolve
0:05.751	finishers	E	templars_verdict	Fluffy_Pillow	5.0/5: 100% holy_power	bloodlust, crusade(4), shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife(2), potion_of_focused_resolve
0:06.575	generators	J	hammer_of_wrath	Fluffy_Pillow	3.0/5: 60% holy_power	bloodlust, crusade, shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife, potion_of_focused_resolve
0:07.398	generators	H	blade_of_justice	Fluffy_Pillow	2.0/5: 40% holy_power	bloodlust, crusade(7), shield_of_vengeance, avengers_might, deadly_navigation, unstable_flames, strife(2), potion_of_focused_resolve
....

**Comment by Hekili** [comment]
Yeah, that's not readable.  But feel free to send me a link to a sim that (1) performs better than the default sim and (2) performs the opener the way you're looking.

**Comment by Retridin** [comment]
Hmm.. Gladly, but how can I do that? How can I change the behaviour of the priorities in the Sim?

And I am not making this up, basically EVERY site  suggest openers the way I tried to write it to you, and not the way Hekili is doing at the moment..
https://www.icy-veins.com/wow/retribution-paladin-pve-dps-rotation-cooldowns-abilities
https://www.wowhead.com/retribution-paladin-rotation-guide
https://www.retpaladin.xyz/ret-paladin-pve-guide

**Comment by Hekili** [comment]
You take the sim profile, refer to the simc wiki for modifying priorities, then you run the sim and see if it did what you wanted (using Raidbots' advanced mode results in a linkable sim).

I understand that guides differ from sims sometimes.  The trick is actually testing the guide vs. sims to show performance differences AND have instructions that sims/addons can interpret.

**Comment by Retridin** [comment]
Linkable sim like this?

https://www.raidbots.com/simbot/report/ekZJ7iWty9P1Jt2xPMMDtu/simc

Ok, I will try to see what I can find from that priority thing, thank you.

**Comment by Hekili** [comment]
Yeah, looking at that sim, it looks like I can get to a similar place:

![image](https://user-images.githubusercontent.com/9438134/78921955-b8dc0780-7a63-11ea-8342-f368baaf44bc.png)

Bear in mind, my Paladin has crappy gear with low haste, so my Blade of Justice hasn't come off cooldown yet (otherwise it'd come up sooner).


For your second request, regarding Righteous Verdict, I'd need to see a sim that uses TV right after Crusade.

**Comment by Retridin** [comment]
I have been trying to figure out that part, that how can I change the "order" that TW would be before HoW..

And one thing I noticed, that if you use Crusade at 5 HP instead of 3, it is slight increase in dps

**Comment by Hekili** [comment]
You can basically change two things:  The position in the priority or the conditions of the entries.

Changing the position can be impactful, because the addon (and your sim) is going to show the first entry that has conditions that pass.

Changing the conditions can be impactful, because if your new conditions for HoW or TV impact when they're shown, then you've changed the priority.

And as simple as that sounds, you can see the complexity of it.  That's why nobody wants to work on priorities.

> And one thing I noticed, that if you use Crusade at 5 HP instead of 3, it is slight increase in dps

It might be, but you have to consider whether it's a worthwhile increase (is it 1%, 0.1%, 0.01%?), whether it's impacted by fight length (delaying Crusade for a few GCDs could mean you lose a few GCDs of Crusade later), whether it's a consistent increase in better/worse gear (does your crit matter, your mastery matter, your haste matter), etc.

Looking at the criteria, it looks like the approach here is to get things on CD, because *time* is a resource as well.  If HoW isn't recharging its CD, you're getting less value out of that time.  So you want to use it ASAP.  Maybe holding it does help.  Maybe not.  If you come up with startling revelations, you should take it to the Paladin discord and share.

**Comment by Retridin** [comment]
Hmmm... I think I got it now. At least it shows in the Sample Sequence Table that TW is cast before HoW and always cast after Crusade. And indeed it seems that at least with ES that is not more efficient. But casting Crusade at 5 HP seems to give consistant dps increase. All this is with Execution sentence -talent.

Basic:
https://www.raidbots.com/simbot/report/6TyVnRduc4hR5Te29UBmrC/simc

TW after Crusade:
https://www.raidbots.com/simbot/report/jZKUHkjPAs6Ru1xUJDYkvK/simc

Crusade at 5HP:
https://www.raidbots.com/simbot/report/weSwKMZCcXe4SRJ9ZPgypz/simc

Using Righteous Verdict -talent doesn't change the situation much, but all are more or less the same.

Thank you for your time and advice, I think I start to get the hang of Hekili better now.
